src/GridFS/WritableStream.php

```diff
@@ -219,12 +231,14 @@ private function abort()
 
     private function fileCollectionInsert()
     {
-        $md5 = hash_final($this->ctx);
-
         $this->file['length'] = $this->length;
-        $this->file['md5'] = $md5;
         $this->file['uploadDate'] = new UTCDateTime;
 
+        if ( ! $this->disableMD5) {
+            $md5 = hash_final($this->hashCtx);
+            $this->file['md5'] = $md5;
```

Comment:
This can probably be consolidated into one line. I likely only broke it up previously to finalize the hash context ASAP and then delayed `$this->file['md5']` assignment to keep it alphabetized with other fields.